addon/utils/utils.js

```diff
@@ -7,18 +7,44 @@ import Ember from 'ember';
 
 const DS = requireModule('ember-data');
 
+const {
+  get
+} = Ember;
+
 export function requireModule(module) {
   return self.requirejs.has(module) ? self.require(module).default : undefined;
 }
 
-export function unwrapString(input) {
-  if (input && input instanceof Ember.Handlebars.SafeString) {
-    return input.toString();
+export function unwrapString(s) {
+  if (s && s instanceof Ember.Handlebars.SafeString) {
+    return s.toString();
+  }
+
+  return s;
+}
+
+export function unwrapProxy(o) {
+  if (o && (o instanceof Ember.ObjectProxy || o instanceof Ember.ArrayProxy)) {
```

Comment:
@aaronbhansen `Ember.Mixin.detect` is private API and can change at any time without warning 😞 .